smart-contracts/contracts/merkle-incentives/src/incentives/mod.rs

```diff
@@ -84,44 +84,60 @@ impl From<Vec<Coin>> for CoinVec {
 }
 
 impl PartialOrd for CoinVec {
-    // todo: test this
+    // This function compares two CoinVec instances (`self` and `other`) based on their coins.
+    // It constructs HashMaps mapping denominations to amounts for both instances.
+    // Then, it iterates over the HashMap of `self` and checks against the HashMap of `other`.
+    // - If a coin in `self` has a corresponding coin in `other`, it compares their amounts.
+    //   - If the amount of the coin in `self` is less than the amount of the corresponding coin in `other`,
+    //     it sets `self_less` to true.
+    //   - If the amount of the coin in `self` is greater than the amount of the corresponding coin in `other`,
+    //     it sets `self_greater` to true.
+    // - If a denomination exists in `other` but not in `self` and the amount is greater than zero, it sets `self_less` to true.
+    // Finally, based on the flags `self_less` and `self_greater`, it returns the ordering:
+    // - If `self_less` is true and `self_greater` is false, it returns `Less`.
+    // - If `self_less` is false and `self_greater` is true, it returns `Greater`.
+    // - If both flags are false, it returns `Equal`.
+    // - If both flags are true, it returns `None`, indicating incomparability
     fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        // if even one in self is greater than other, this should return greater
-        for coin in &self.0 {
-            if let Some(other_coin) = other.0.iter().find(|c| c.denom == coin.denom) {
-                if coin.amount > other_coin.amount {
-                    return Some(std::cmp::Ordering::Greater);
+        let self_map: std::collections::HashMap<_, _> = self
```

Comment:
Theoretically, iterating through lists with less than a couple thousand items is faster than a hashmap. However that does not matter much right now with gas prices. No action needed but we should use `Vec` in the package

Reply:
Added todo for later